simulator/build.gradle.kts

```diff
@@ -94,22 +94,26 @@ tasks.named("processResources") { dependsOn(tasks.named("untarTestBlockStream"))
 
 tasks.named("sourcesJar") { dependsOn(tasks.named("untarTestBlockStream")) }
 
-// Task to prepare and run Docker container
-tasks.register("startDockerContainer") {
-    description = "Prepare and run the simulator containers in publisher and consumer mode"
-    group = "docker"
-
-    doLast {
-        exec { commandLine("./docker/prepare-docker.sh") }
-
-        exec {
-            workingDir("docker")
-            commandLine("sh", "-c", "docker compose -p simulator up --build -d")
-        }
+// Vals
+val dockerProjectRootDirectory: Directory = layout.projectDirectory.dir("docker")
+val dockerBuildRootDirectory: Directory = layout.buildDirectory.dir("docker").get()
+
+// Docker related tasks
+val copyDockerFolder: TaskProvider<Copy> =
+    tasks.register<Copy>("copyDockerFolder") {
+        description = "Copies the docker folder to the build root directory"
+        group = "docker"
+
+        from(dockerProjectRootDirectory)
+        into(dockerBuildRootDirectory)
+    }
 
-        exec { commandLine("./docker/cleanup-docker.sh") }
+val startDockerContainer: TaskProvider<Exec> =
+    tasks.register<Exec>("startDockerContainer") {
+        description = "Creates and starts the docker image of the Block Stream Simulator"
+        group = "docker"
 
-        println("✅ Docker simulator is now running!")
-        println("🧹 Build artifacts have been cleaned up")
+        dependsOn(copyDockerFolder, tasks.assemble)
+        workingDir(dockerBuildRootDirectory)
+        commandLine("sh", "-c", "./prepare-docker.sh")
     }
```

Comment:
Let's also add the `startDockerContainerDebug` and `stopDockerContainer` tasks as it is in the server.

Reply:
Added `stopDockerContainer`, but skipped `startDockerContainerDebug` for two reasons:
1. Currently we are starting two instances of the same application, therefore have to take care of mapping the ports correcty.
2. There is no request for this now, don't want to add unnecessary things for them to never be used. When there is a need, we can add it easily.